DigitalLearningSolutions.Web/Controllers/TrackingSystem/Delegates/GroupDelegatesController.cs

```diff
@@ -0,0 +1,217 @@
+            IUserService userService,
+            CentreCustomPromptHelper centreCustomPromptHelper,
+            IGroupsService groupsService
+        )
+        {
+            this.centreCustomPromptHelper = centreCustomPromptHelper;
+            this.jobGroupsService = jobGroupsService;
+            this.userService = userService;
+            this.groupsService = groupsService;
+        }
+
+        [Route("{page:int=1}")]
+        public IActionResult Index(int groupId, int page = 1)
+        {
+            var centreId = User.GetCentreId();
+            var groupName = groupsService.GetGroupName(groupId, centreId);
+
+            if (groupName == null)
+            {
+                return NotFound();
+            }
+
+            var groupDelegates = groupsService.GetGroupDelegates(groupId);
+
+            var model = new GroupDelegatesViewModel(groupId, groupName, groupDelegates, page);
+
+            return View(model);
+        }
+        
+        [HttpGet("Add/SelectDelegate/{page=1:int}")]
+        public IActionResult SelectDelegate(
+            int groupId,
+            string? searchString = null,
+            string? filterBy = null,
+            string? filterValue = null,
+            int page = 1
+        )
+        {
+            filterBy = FilteringHelper.GetFilterBy(
+                filterBy,
+                filterValue,
+                Request,
+                AddGroupDelegateCookieName
+            );
+
+            var centreId = User.GetCentreId();
+            var jobGroups = jobGroupsService.GetJobGroupsAlphabetical().ToList();
+            var customPrompts = centreCustomPromptHelper.GetCustomPromptsForCentre(centreId).ToList();
+            var delegateUsers = userService.GetDelegatesNotRegisteredForGroupByGroupId(groupId, centreId);
+            var groupName = groupsService.GetGroupName(groupId, centreId);
+
+            var model = new AddGroupDelegateViewModel(
+                delegateUsers,
+                jobGroups,
+                customPrompts,
+                page,
+                groupId,
+                groupName!,
+                searchString,
+                filterBy
+            );
+
+            Response.UpdateOrDeleteFilterCookie(AddGroupDelegateCookieName, filterBy);
+            return View(model);
+        }
+
+        [HttpPost("Add/{delegateId:int}")]
+        public IActionResult AddDelegate(int groupId, int delegateId)
+        {
+            var delegateUser = userService.GetDelegateUserById(delegateId);
+            if (delegateUser == null)
+            {
+                return NotFound();
+            }
+
+            var adminId = User.GetAdminId();
+
+            var newDetails = new MyAccountDetailsData(
+                null,
+                delegateUser.Id,
+                string.Empty,
+                delegateUser.FirstName!,
+                delegateUser.LastName!,
+                delegateUser.EmailAddress!,
+                null
+            );

```

Comment:
I think this can be an `AccountDetailsData` which only contains the fields we need, instead of having to set some as null.

Reply:
`AccountDetailsData` is an abstract class so we can use it directly. Although in saying that this code has now gone from here.